app/queries/servers/system.ts

```diff
@@ -365,6 +410,29 @@ export async function prepareCommonSystemValues(
     }
 }
 
+export async function prepareDataRetentionPolicies(
+    operator: ServerDataOperator, {globalPolicy, teamPolicies, channelPolicies}: DataRetentionPoliciesRequest): Promise<SystemModel[]> {
+    try {
+        const systems: IdValue[] = [{
+            id: SYSTEM_IDENTIFIERS.DATA_RETENTION_POLICIES,
+            value: JSON.stringify(globalPolicy || {}),
```

Comment:
no need to stringify here

Reply:
Also there is nothing here checking if the values are different before adding/updating them in the database